src/H5Shyper.c

```diff
@@ -3282,6 +3282,14 @@ H5S__hyper_is_valid(const H5S_t *space)
 
     assert(space);
 
+    /* Check if dataspace has scalar or null extent, which are
+     * both unsupported by hyperslab selections
+     */
+    if (H5S_SCALAR == H5S_GET_EXTENT_TYPE(space))
+        HGOTO_DONE(false);
+    if (H5S_NULL == H5S_GET_EXTENT_TYPE(space))
```

Comment:
Added this logic here for the future. I wanted to add an `if (!H5S_IS_VALID())` check to the top-level `H5Sget_select_hyper_blocklist()`, but then I got testing failures in a test that calls the function on an unlimited selection that is converted to a limited selection. I suspect this is another bug.